app/controllers/application_controller.rb

```diff
@@ -1,2 +1,5 @@
 class ApplicationController < ActionController::Base
+  def current_user
```

Comment:
- [ ] Dear @FaranoshAmini, Please consider adding an action for `new` and `create` functionality of the `Post_controller`, I can see you have not made any changes to the Post_controller, but according to the requirement the posts controller should have new and create actions. Please add these actions.
- [ ] Also, there should be a form in the views/posts/new.html.erb file which accepts input for creating posts.